app/packages/core/src/components/ColorModal/LabelTag.tsx

```diff
@@ -193,20 +190,35 @@ const LabelTag: React.FC = () => {
           >
             {/* set attribute value - color */}
             <Checkbox
-              name={`Use custom colors for specific field values`}
+              name={`Use custom colors for specific label tag values`}
               value={state.useLabelColors}
               setValue={(v: boolean) => {
-                const copy = cloneDeep(labelTags);
-                copy.valueColors = v
-                  ? [{ value: "", color: defaultColor }]
-                  : [];
-
-                setColorScheme({
-                  colorPool,
-                  fields,
-                  labelTags: copy,
+                setSetting((cur) => {
+                  if (!cur) {
+                    cur = { valueColors: [] };
+                  }
+
+                  if (!cur?.valueColors?.length && v) {
+                    cur = {
+                      ...cur,
+                      valueColors: [
+                        {
+                          value: "",
+                          color:
+                            colorPool[
+                              Math.floor(Math.random() * colorPool.length)
+                            ],
+                        },
+                      ],
```

Comment:
nit: make a utility function to return a new valueColors array to keep this simple and readable

Reply:
I also see in lin 104 in app/packages/core/src/components/ColorModal/colorPalette/AttributeColorSetting.tsx which a utility function should help refactor